spi/src/main/kotlin/org/opensearch/alerting/spi/RemoteMonitorRunner.kt

```diff
@@ -49,7 +49,7 @@ open class RemoteMonitorRunner {
     }
 
     open fun getFanOutAction(): String {
-        return DocLevelMonitorFanOutAction.NAME
+        throw UnsupportedOperationException("Fan out action needs to be implemented by remote monitor.");
```

Comment:
Why is this being thrown as an exception now instead of returning the name?

Reply:
this class needs to be implemented and registered at
`public Map<String, RemoteMonitorRunner> getMonitorTypesToMonitorRunners()` in Plugin

should not have a default fan out action. needs to be specified by the new monitor created by extending this class